app/models/manifest_uploader.rb

```diff
@@ -92,4 +96,14 @@ def check_if_any_labwares_are_locations
       errors.add(:base, "Labware barcodes cannot be the same as an existing location barcode. Please review and remove incorrect labware barcodes")
     end
   end
+
+  def check_user
```

Comment:
What happens here if the user is nil. Will it return it as valid? If in the controller or the form you run User.find_by_code irrespective of whether it is present it will return a guest user which will be invalid.

Reply:
If the user is nil, the error `undefined method `guest?' for nil:NilClass` is thrown. But like you say, the `find_by_code` will always return atleast a guest so this should never happen. 
I have moved the set and get of current user into the manifest as it was common behavior like you pointed out. The user does have to be set before validation so thinking I'll set the instance variable `@current_user` in the `check_user`